src/librustc/ty/relate.rs

```diff
@@ -431,6 +431,9 @@ pub fn super_relate_tys<R: TypeRelation<'tcx>>(
             let t = relation.relate(&a_t, &b_t)?;
             match relation.relate(&sz_a, &sz_b) {
                 Ok(sz) => Ok(tcx.mk_ty(ty::Array(t, sz))),
+                // FIXME(lazy-normalization) Implement improved diagnostics for mismatched array
```

Comment:
```suggestion
                // FIXME(lazy_normalization) Implement improved diagnostics for mismatched array
```
Let's use the same name as the feature gate / option for consistency and searchability.